js/data/program_configuration.js

```diff
@@ -210,7 +210,8 @@ function getPaintAttributeValue(attribute, layer, globalProperties, featurePrope
 }
 
 function normalizePaintAttribute(attribute, layer) {
-    const name = attribute.property.replace(`${layer.type}-`, '').replace(/-/g, '_');
+    const layerTypePrefix = layer.type === 'symbol' ? /text-|icon-/ : `${layer.type}-`;
+    const name = attribute.property.replace(layerTypePrefix, '').replace(/-/g, '_');
```

Comment:
> Can you think of any cases where this magic behaviour might bite us later?

@lucaswoj not offhand, but I'm 💯 with you in being uneasy / less than satisfied with this magic.

>  I'd be interested to see if there was a way to specify this behaviour explicitly within SymbolBucket instead.

👍 I'm thinking we can allow the bucket to explicitly define `attribute.name`, and then have `normalizePaintAttribute` just use that if it's there, instead of computing the attribute name from the paint property name.